### ISSUE TITLE
Glitchy screen tearing artifacts in ffmpeg's x11grab screen capture

### ISSUE THREAD

**Issue by tm512** [yshui/compton#943]
### Platform
Void Linux x86_64 w/ kernel 6.0.9

### GPU, drivers, and screen setup
GTX 750 Ti w/ nvidia proprietary 515.86.01

`glxinfo -B`:
```text
name of display: :0
display: :0  screen: 0
direct rendering: Yes
Memory info (GL_NVX_gpu_memory_info):
    Dedicated video memory: 2048 MB
    Total available memory: 2048 MB
    Currently available dedicated video memory: 1674 MB
OpenGL vendor string: NVIDIA Corporation
OpenGL renderer string: NVIDIA GeForce GTX 750 Ti/PCIe/SSE2
OpenGL core profile version string: 4.6.0 NVIDIA 515.86.01
OpenGL core profile shading language version string: 4.60 NVIDIA
OpenGL core profile context flags: (none)
OpenGL core profile profile mask: core profile

OpenGL version string: 4.6.0 NVIDIA 515.86.01
OpenGL shading language version string: 4.60 NVIDIA
OpenGL context flags: (none)
OpenGL profile mask: (none)

OpenGL ES profile version string: OpenGL ES 3.2 NVIDIA 515.86.01
OpenGL ES profile shading language version string: OpenGL ES GLSL ES 3.20
```

relevant `xorg.conf` section:
```text
Section "Device"
	Identifier "nvidia0"
	Driver "nvidia"
	Option "TripleBuffer" "True"
	Option "UseEvents" "True"
EndSection
```

### Environment
I'm using an old patched build of dwm from ca. 2015 on top of X.org Server 21.1.4. I could try making a new build of dwm, if that would make a meaningful difference here. I've just put it off for this long since I'd have to basically start from scratch with patching.

### picom version
vgit-8a373
<details>
<summary>Diagnostics</summary>
**Version:** vgit-8a373

### Extensions:

* Shape: Yes
* XRandR: Yes
* Present: Present

### Misc:

* Use Overlay: No
  (Another compositor is already running)
* Config file used: None

### Drivers (inaccurate):

NVIDIA

### Backend: glx

* Driver vendors:
 * GLX: NVIDIA Corporation
 * GL: NVIDIA Corporation
* GL renderer: NVIDIA GeForce GTX 750 Ti/PCIe/SSE2

### Backend: egl

* Driver vendors:
 * EGL: NVIDIA
 * GL: NVIDIA Corporation
* GL renderer: NVIDIA GeForce GTX 750 Ti/PCIe/SSE2
</details>

### Configuration:
Not using a configuration file, just launching picom as follows from `.xinitrc`:

```text
picom -CGb --backend glx --vsync --glx-no-stencil --xrender-sync-fence --glx-no-rebind-pixmap --no-fading-openclose
```

### Steps of reproduction
Unfortunately, I don't know how to reliably reproduce this. It's not a constant thing, and when it does show up, it doesn't seem to have any rhyme or reason to it.

### Expected behavior
x11grab's output (running at screen's native FPS) should exactly reflect what I'm seeing on screen

### Current Behavior
Capture output regularly has screen tearing that often isn't a simple shear like what you'd get from video not being synced to the screen's refresh rate. None of this is visible to me unless I'm actively monitoring the actual screen capture. It might have some relation to #123 since the positioning of the glitching and its shape seem kinda similar.

I've uploaded a small clip from a larger capture here that highlights the issue: https://i.imgur.com/ZcsurxD.mp4

**Comment by yshui** [comment]
hmm are you positive that this does not happen if picom is not running?

nonetheless, i think it's a good idea to report this to nvidia as well.

**Comment by Monsterovich** [comment]
@tm512 Why not use OBS, which has support for nvenc and recording individual windows?

**Comment by tm512** [comment]
@yshui I haven't really done much testing without, as my desktop is pretty much intolerable to use without picom running. The ever-present screen tearing that results from picom not running could also make it harder to spot this issue. I suppose I could test with "Force Composition Pipeline" enabled, and picom disabled, but the last time I tried FCP/FFCP, back on driver 470.x or so, it had a brutal performance hit, significantly worse than picom's vsync option ever causes. If it happens to be a usable solution here on 515.86, I'll test and report back.

I also had very similar tearing artifacts (albeit differently shaped and positioned) back when I was using an Intel iGPU along with one of the last versions of compton that Void packaged (an example screenshot of that is https://i.imgur.com/xt8szD5.png). That seemed to be resolved by upgrading to picom, but the similarity is partially why I've jumped to the conclusion that picom has something to do with my current issue here on nvidia.

@Monsterovich I'm not exactly lacking support for those features, but I mostly stick with ffmpeg because I already have a working setup, and it's nice to just have a small terminal running my capture script where I can monitor things, I simply don't have the screen real-estate to have a full OBS window visible along with everything else. I kinda doubt that it will make much of a difference anyway, since I think it's going through the same X11 API to grab the screen contents.
